autoscale/src/main/java/com/sequenceiq/periscope/monitor/handler/PersistRejectedThreadExecutionHandler.java

```diff
@@ -34,6 +34,9 @@ public class PersistRejectedThreadExecutionHandler extends AbortPolicy {
     public PersistRejectedThreadExecutionHandler() {
         try {
             callableInFutureTask = FutureTask.class.getDeclaredField("callable");
+            // TODO: rewrite it!!!
+            // It is an ugly and fragile hack and it needs opening Java internal modules with
+            // --add-opens java.base/java.util.concurrent=ALL-UNNAMED
```

Comment:
please ensure we have jira task for it

Reply:
+1. Actually, our initial plan was indeed raising follow-up tickets for all the hacks applied here. You can find the complete list in the associated [wiki](https://cloudera.atlassian.net/wiki/spaces/ENG/pages/10335519964/HOWTO+-+Cloudbreak+development+on+Java17).

Reply:
Story created under [epic](https://jira.cloudera.com/browse/CB-21717)